pkg/api/v2/types.go

```diff
@@ -1062,55 +1059,13 @@ type GKEUpgradeSettings struct {
 // swagger:model VirtualMachineInstancePresetList
 type VirtualMachineInstancePresetList []VirtualMachineInstancePreset
 
-// Need to copy the following type to avoid a collision on Resources
-// between kubevirtv1.ResourceRequirements and corev1.ResourceRequirements used in different part of the API.
-type DomainSpec struct {
-	// Resources describes the Compute Resources required by this vmi.
-	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
-	// CPU allow specified the detailed CPU topology inside the vmi.
-	// +optional
-	CPU *kubevirtv1.CPU `json:"cpu,omitempty"`
-	// Memory allow specifying the VMI memory features.
-	// +optional
-	Memory *kubevirtv1.Memory `json:"memory,omitempty"`
-	// Machine type.
-	// +optional
-	Machine *kubevirtv1.Machine `json:"machine,omitempty"`
-	// Firmware.
-	// +optional
-	Firmware *kubevirtv1.Firmware `json:"firmware,omitempty"`
-	// Clock sets the clock and timers of the vmi.
-	// +optional
-	Clock *kubevirtv1.Clock `json:"clock,omitempty"`
-	// Features like acpi, apic, hyperv, smm.
-	// +optional
-	Features *kubevirtv1.Features `json:"features,omitempty"`
-	// Devices allows adding disks, network interfaces, and others
-	Devices kubevirtv1.Devices `json:"devices"`
-	// Controls whether or not disks will share IOThreads.
-	// Omitting IOThreadsPolicy disables use of IOThreads.
-	// One of: shared, auto
-	// +optional
-	IOThreadsPolicy *kubevirtv1.IOThreadsPolicy `json:"ioThreadsPolicy,omitempty"`
-	// Chassis specifies the chassis info passed to the domain.
-	// +optional
-	Chassis *kubevirtv1.Chassis `json:"chassis,omitempty"`
-}
-
-type VirtualMachineInstancePresetSpec struct {
-	// Selector is a label query over a set of VMIs.
-	// Required.
-	Selector metav1.LabelSelector `json:"selector"`
-	// Domain is the same object type as contained in VirtualMachineInstanceSpec
-	Domain *DomainSpec `json:"domain,omitempty"`
-}
-
 // VirtualMachineInstancePreset represents a KubeVirt Virtual Machine Instance Preset
 // swagger:model VirtualMachineInstancePreset
 type VirtualMachineInstancePreset struct {
-	apiv1.ObjectMeta `json:",inline"`
-	// VirtualMachineInstance Spec contains the VirtualMachineInstance specification.
-	Spec VirtualMachineInstancePresetSpec `json:"spec,omitempty" valid:"required"`
+	Name      string `json:"name,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
Is the namespace required, will it be displayed in the UI?

Reply:
That would be great yes, as it's a namespace resource.

Reply:
Okay 👍🏻